### PR TITLE
MVP for Spine client with C++

### PR DESCRIPTION
**Pull request by vladimir-moiseiev**
A MVP version of Spine client library for C++.
Should be build using CMake 3.9 version and higher.
Dependencies:
- Spine proto files [spine-proto.jar](http://maven.teamdev.com/repository/spine-snapshots/io/spine/spine-client/0.10.1-SNAPSHOT/spine-client-0.10.1-20171019.131949-2-proto.jar)
- Google Protobuf 3.x [link](https://developers.google.com/protocol-buffers/)
- Google gRPC [link](https://grpc.io)
- Boost [link](http://www.boost.org/)
- Poco [link](https://pocoproject.org/)
- GTest [link](https://github.com/google/googletest)

Artifacts for this library are the header files and spine-*.a or spine-*.lib static library.
Header files can be found in the "include" folder.

### CODE REVIEW

**Comment by armiol** [comment]
@vladimir-moiseiev can we please address the following points before merging?
* create a description for the PR — as PRs serve as documentation on these early project stages;
* configure Travis to do nothing and have the build passed (by default it performs Ruby `rake` build and goes crazy :) );
* rename "spine" to  "Spine" in the PR title.

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+echo "GRPC in $1"
+
+mkdir -p .build
+cd .build
+
+cmake -DGRPC_INSTALL_DIR=$1 ..
+cmake --build . -- -j5
```
Comment (armiol) [inline build.sh]: GitHub detects improper line endings in several files within this PR. Please fix those.

We address CRLF issues by using `core.autocrlf` Git configuration property. Or — as an alternative — use the IDEA (or CLion) setting.

File: examples/hello_world/cpp-client-reader/main_receiver.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    if (response.messages_size() > 0)
+    {
+        const Any& any = response.messages(0);
+        ::helloworld::LastHello last_hello;
+        any.UnpackTo(&last_hello);
+        std::string username = last_hello.username();
+    }
+    return 0;
+}
+
```
Comment (armiol) [inline examples/hello_world/cpp-client-reader/main_receiver.cc]: Please remove the redundant line.

File: examples/hello_world/cpp-client-sender/main_sender.cc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    std::unique_ptr<CommandService::Stub> stub = CommandService::NewStub(channel);
+
+    spine::core::Ack response;
+    grpc::ClientContext client_context;
+    stub->Post(&client_context, *command, &response);
+
+    return 0;
+}
+
```
Comment (armiol) [inline examples/hello_world/cpp-client-sender/main_sender.cc]: Please remove this redundant line as well.

File: src/command_factory.cc [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    return command_context;
+}
+
+CommandContext* make_command_context(const std::unique_ptr<ActorContext>& actor_context, const int version)
+{
+    CommandContext* context = make_command_context(actor_context);
+    context->set_target_version(version);
+    return context;
+}
+
```
Comment (armiol) [inline src/command_factory.cc]: Please remove the redundant lines.

File: test/type_url_test.cc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+TEST(TypeUrlShould, ThrowAtParseWithInvalidParams)
+{
+    ASSERT_FALSE(TypeUrl::parse(""));
+    ASSERT_FALSE(TypeUrl::parse("Type"));
+    ASSERT_FALSE(TypeUrl::parse("Type/"));
+    ASSERT_FALSE(TypeUrl::parse("/"));
+    ASSERT_FALSE(TypeUrl::parse("/sadfsad"));
+}
+
+//TEST(TypeUrlShould, ThrowAtCreateWithInvalidParams)
```
Comment (armiol) [inline test/type_url_test.cc]: Can we please remove the commented portion or have a comment that it is needed, but, say, "commented until ...".

**Comment by vladimir-moiseiev** [comment]
@armiol PTAL
About the merge, i'd prefer to squash all commits for this PR, if possible in this repo.

File: build.sh (hunk 1 above)
Comment (vladimir-moiseiev) [reply build.sh]: Removed the scripts, until they are ready.

**Pull request merged**